cmd/git/main.go

```diff
@@ -82,6 +83,7 @@ func init() {
 
 	// Mostly internal flag
 	pflag.BoolVar(&flagValues.skipValidation, "skip-validation", false, "skip pre-requisite validation")
+	pflag.BoolVar(&flagValues.gitURLRewrite, "git-url-rewrite", false, "set Git config to use url-insteadOf setting based on Git repository URL")
```

Comment:
hmm ... when are we going to "bite the bullet" and start controlling behavior like this via settings in config maps vs. start up flags

Reply:
I know, I know. Do not have a good answer for that. With a config map, I think there was the discussion that this will lead to higher expectations, for example that an operator will likely assume that a change in the config map will immediately be used by the controller.

Reply:
Imo the command line flags in step containers like git or mutate-image will never go away because the BuildRun pod in the user's namespace will never access a ConfigMap that represents the Shipwright configuration (in the shipwright namespace). The other question is when the environment variables on the controller yaml will get replaced by a ConfigMap. Imo this can happen at any time, though as long as it has not been done, we should not stop any feature development that by coincident needs a configuration.

Reply:
a config map with the config can be created in the user's namespace by the shipwright controller, based on config (including config maps) that drive the behavior of the controller, including how it sets up the build run pod, so that the user namespace config map drives the behavior of the build run pod

such a pattern is employed in openshift builds today

as our discussion here reflects, a separate effort, including most likely an EP, should be pursued ... probably discussion in community or grooming meetings to collectively decide when we will get to this 

@HeavyWombat efforts here just remind us we keep putting this off :-)

Reply:
@gabemontero The config map idea has an issue as far as I know. We just need someone to find time and the energy to do it. Independent of that, are you good to let this PR go into the `main` branch? Thanks.

Reply:
@HeavyWombat yeah the config map idea has been discussed, but I could not find a tracking issue (via https://github.com/shipwright-io/build/issues?q=is%3Aissue+is%3Aopen+config+in%3Atitle)

but we don't need to sort that out here

so regardless , yes, as I said before, I'm fine with the PR moving forward independent of that.

that said, non-shipwright items have popped for me and unfortunately it will be at least until second half of next week before I can do a legitimate code review (vs. the cursory scan I did last week)

if you don't want to wait that long of course assign anybody from the @sbose78 @adambkaplan @imjasonh @coreydaley @otaviof list of hatters and they can pick up

thanks